### ISSUE TITLE
Issue when migrating to SDK V2

### ISSUE THREAD

**Issue by kaissun** [aws/aws-sdk-java-v2#2920]
### Describe the issue

Hello team, 

I am migrating my solution using SDK V2 and I followed like described in the [article](https://aws.amazon.com/fr/blogs/developer/introducing-amazon-s3-transfer-manager-in-the-aws-sdk-for-java-2-x/). 
It is working well using SDK V1 but when I am trying the apply the changes and using the same properties, I faced this issue: 

software.amazon.awssdk.core.exception.SdkClientException: software.amazon.awssdk.crt.s3.CrtS3RuntimeException: Host name was invalid for dns resolution. AWS_IO_DNS_INVALID_NAME(1059)

Do you have any idea that causes to discover the problem with hostname? 
I am using the credentials by the way and I already applied the exact region related as well.

Thanks in advance.

### Steps to Reproduce

Use the same steps described in this article https://aws.amazon.com/fr/blogs/developer/introducing-amazon-s3-transfer-manager-in-the-aws-sdk-for-java-2-x/

### Current behavior

Error is occurred: 

software.amazon.awssdk.core.exception.SdkClientException: software.amazon.awssdk.crt.s3.CrtS3RuntimeException: Host name was invalid for dns resolution. AWS_IO_DNS_INVALID_NAME(1059)

### AWS Java SDK version used

2.17.16-PREVIEW

### JDK version used

java version "11.0.3" 2019-04-16 LTS

### Operating System and version

linux

**Comment by debora-ito** [comment]
Hi @kaissun thank you for reaching out.

Can you provide a code sample we can use to reproduce?
And can you share the full stacktrace of the error?

**Comment by kaissun** [comment]
Hello, 

Sorry for the late reply. 
Before giving the sample code and the error stack trace, I explain briefly the context that I am working with. It is about an investigation with the maximum upload speed when using the S3 multi upload feature. 

I contacted the support team of AWS and they said that the maximum upload speed for a file that exceeds 150mo will be 150mo/s. And if I would like to upload big files, they recommend me to use the S3 multi upload feature. 
I tried with the SDK V1 and I reached only maximum 40mo/s upload speed. 
And now I am trying with the SDK V2, hope that it was optimized. 
Note that I have to pass always by a proxy.

I will present fist the sample code using java and then I will share the full stacktrace of the error. 

```
System.setProperty("http.proxyHost", proxyHost);
System.setProperty("http.proxyPort", proxyPort+"");
AwsBasicCredentials awsCreds = AwsBasicCredentials.create(accessKeyId, secretAccessKey);
S3TransferManager transferManager =
        S3TransferManager.builder()
                        .s3ClientConfiguration(cfg -> cfg.credentialsProvider(StaticCredentialsProvider.create(awsCreds))
                                                        .region(Region.EU_WEST_1)
                                                        .targetThroughputInGbps(20.0)
                                                        .minimumPartSizeInBytes((long) 10 * mB))
                        .build();

Upload upload = transferManager.upload(b -> b.source(Paths.get(fileToUploadPath))
                        .putObjectRequest(req -> req.bucket(bucketName)
                                                    .key(key)));
upload.completionFuture().join();
```

And about the stacktrace si like the following:

```
Exception in thread "main" java.util.concurrent.CompletionException: software.amazon.awssdk.core.exception.SdkClientException: software.amazon.awssdk.crt.s3.CrtS3RuntimeException: Host name was invalid for dns resolution. AWS_IO_DNS_INVALID_NAME(1059)
        at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:331)
        at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:346)
        at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:632)
        at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506)
        at java.base/java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:2088)
        at software.amazon.awssdk.transfer.s3.internal.DefaultS3CrtAsyncClient.lambda$putObject$3(DefaultS3CrtAsyncClient.java:137)
        at java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:859)
        at java.base/java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:837)
        at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506)
        at java.base/java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:2088)
        at com.amazonaws.s3.S3NativeClient$3.onFinished(S3NativeClient.java:271)
        at software.amazon.awssdk.crt.s3.S3MetaRequestResponseHandlerNativeAdapter.onFinished(S3MetaRequestResponseHandlerNativeAdapter.java:19)
Caused by: software.amazon.awssdk.core.exception.SdkClientException: software.amazon.awssdk.crt.s3.CrtS3RuntimeException: Host name was invalid for dns resolution. AWS_IO_DNS_INVALID_NAME(1059)
        at software.amazon.awssdk.core.exception.SdkClientException$BuilderImpl.build(SdkClientException.java:98)
        at software.amazon.awssdk.core.exception.SdkClientException.create(SdkClientException.java:43)
        at software.amazon.awssdk.transfer.s3.internal.CrtErrorHandler.transformException(CrtErrorHandler.java:56)
        at software.amazon.awssdk.transfer.s3.internal.DefaultS3CrtAsyncClient.lambda$putObject$3(DefaultS3CrtAsyncClient.java:138)
        ... 6 more
Caused by: java.util.concurrent.CompletionException: software.amazon.awssdk.crt.s3.CrtS3RuntimeException: Host name was invalid for dns resolution. AWS_IO_DNS_INVALID_NAME(1059)
        at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:331)
        at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:346)
        at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:632)
        ... 4 more
Caused by: software.amazon.awssdk.crt.s3.CrtS3RuntimeException: Host name was invalid for dns resolution. AWS_IO_DNS_INVALID_NAME(1059)
        at com.amazonaws.s3.S3NativeClient$3.onFinished(S3NativeClient.java:263)
        ... 1 more

```

And I also used the following dependencies for the maven project:

```
        <dependency>
            <groupId>software.amazon.awssdk</groupId>
            <artifactId>s3-transfer-manager</artifactId>
            <version>2.17.16-PREVIEW</version>
        </dependency>
        <dependency>
            <groupId>software.amazon.awssdk</groupId>
            <artifactId>apache-client</artifactId>
            <version>2.17.99</version>
          </dependency>
```

thanks in advance for your contribution

**Comment by debora-ito** [comment]
@kaissun my guess would be that the SDK is not able to reach the hostname due to some proxy configuration.

The TransferManager got some changes recently as part of a refactor, can you try the latest version and see if you still experience the issue? The changes were released in version `2.17.119-PREVIEW`.
